src/appservice/MjolnirManager.ts

```diff
@@ -0,0 +1,224 @@
+        );
+        this.mjolnirs.set(await client.getUserId(), managedMjolnir);
+        return managedMjolnir;
+    }
+
+    /**
+     * Gets a mjolnir for the corresponding mxid that is owned by a specific user.
+     * @param mjolnirId The mxid of the mjolnir we are trying to get.
+     * @param ownerId The owner of the mjolnir. We ask for it explicitly to not leak access to another user's mjolnir.
+     * @returns The matching managed mjolnir instance.
+     */
+    public getMjolnir(mjolnirId: string, ownerId: string): ManagedMjolnir|undefined {
+        const mjolnir = this.mjolnirs.get(mjolnirId);
+        if (mjolnir) {
+            if (mjolnir.ownerId !== ownerId) {
+                throw new Error(`${mjolnirId} is owned by a different user to ${ownerId}`);
+            } else {
+                return mjolnir;
+            }
+        } else {
+            return undefined;
+        }
+    }
+
+    /**
+     * Find all of the mjolnirs that are owned by this specific user.
+     * @param ownerId An owner of multiple mjolnirs.
+     * @returns Any mjolnirs that they own.
+     */
+    public getOwnedMjolnirs(ownerId: string): ManagedMjolnir[] {
+        // TODO we need to use the database for this but also provide the utility
+        // for going from a MjolnirRecord to a ManagedMjolnir.
+        // https://github.com/matrix-org/mjolnir/issues/409
+        return [...this.mjolnirs.values()].filter(mjolnir => mjolnir.ownerId !== ownerId);
+    }
+
+    public onEvent(request: Request<WeakEvent>, context: BridgeContext) {
+        // TODO We need a way to map a room id (that the event is from) to a set of managed mjolnirs that should be informed.
+        // https://github.com/matrix-org/mjolnir/issues/412
+        [...this.mjolnirs.values()].forEach((mj: ManagedMjolnir) => mj.onEvent(request));
+    }
+
+    /**
+     * provision a new mjolnir for a matrix user.
+     * @param requestingUserId The mxid of the user we are creating a mjolnir for.
+     * @returns The matrix id of the new mjolnir and its management room.
+     */
+    public async provisionNewMjolnir(requestingUserId: string): Promise<[string, string]> {
+        const access = this.accessControl.getUserAccess(requestingUserId);
+        if (access.outcome !== Access.Allowed) {
+            throw new Error(`${requestingUserId} tried to provision a mjolnir when they do not have access ${access.outcome} ${access.rule?.reason ?? 'no reason specified'}`);
+        }
+        const provisionedMjolnirs = await this.dataStore.lookupByOwner(requestingUserId);
+        if (provisionedMjolnirs.length === 0) {
+            const mjolnirLocalPart = `mjolnir_${randomUUID()}`;
+            const [mjolnirUserId, mjolnirClient] = await this.makeMatrixClient(mjolnirLocalPart);
+
+            const managementRoomId = await mjolnirClient.createRoom({
+                preset: 'private_chat',
+                invite: [requestingUserId],
+                name: `${requestingUserId}'s mjolnir`
```

Comment:
Also, note to self: we'll need to think of how we're going to internationalize Mjölnir, one of these days.